lib/universal/mapper.js

```diff
@@ -0,0 +1,785 @@
+ *    - v
+ *
+ * @param {Track} track
+ * @param {Object} settings
+ * @return {Object}
+ */
+
+exports.addedProduct = function(track, settings){
+  return extend(
+    {
+      ea: track.event(),
+      ec: track.field('category') || 'EnhancedEcommerce',
+      pa: 'add',
+      t: 'event'
+    },
+    createCommonGAForm(track, settings),
+    formatPage(track),
+    formatProduct(track.properties())
+  );
+};
+
+/**
+ * Map Removed Product.
+ *
+ * https://developers.google.com/analytics/devguides/collection/protocol/v1/devguide#enhancedecom
+ * https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters
+ *
+ * Required properties:
+ *    - cid
+ *    - ea
+ *    - ec
+ *    - pa
+ *    - pr1ca
+ *    - pr1id
+ *    - pr1nm
+ *    - pr1pr
+ *    - pr1qty
+ *    - t
+ *    - tid
+ *    - v
+ *
+ * @param {Track} track
+ * @param {Object} settings
+ * @return {Object}
+ */
+
+exports.removedProduct = function(track, settings){
+  return extend(
+    {
+      ea: track.event(),
+      ec: track.field('category') || 'EnhancedEcommerce',
+      pa: 'remove',
+      t: 'event'
+    },
+    createCommonGAForm(track, settings),
+    formatPage(track),
+    formatProduct(track.properties())
+  );
+};
+
+/**
+ * Map Started Order.
+ *
+ * https://developers.google.com/analytics/devguides/collection/protocol/v1/devguide#enhancedecom
+ * https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters
+ *
+ * Required parameters:
+ *    - cid      (set in createcommonGAForm)
+ *    - tid      (set in createcommonGAForm)
+ *    - v        (set in createcommonGAForm)
+ *
+ * @param {Track} track
+ * @param {Object} settings
+ * @return {Object}
+ */
+
+exports.startedOrder = function(track, settings){
+  return exports.viewedCheckoutStep(track, settings);
+};
+
+/**
+ * Map Updated Order.
+ *
+ * https://developers.google.com/analytics/devguides/collection/analyticsjs/enhanced-ecommerce#checkout-steps
+ * https://developers.google.com/analytics/devguides/collection/protocol/v1/devguide#enhancedecom
+ * https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters
+ *
+ * Required parameters:
+ *    - cid      (set in createcommonGAForm)
+ *    - tid      (set in createcommonGAForm)
+ *    - v        (set in createcommonGAForm)
+ *    - pa       ('checkout') ?
+ *    - pr{x}{x} (.properties.products[x].z)
+ *    - t        ('pageview') TODO: right?
+ *
+ * @param {Track} track
+ * @param {Object} settings
+ * @return {Object}
+ */
+
+exports.updatedOrder = function(track, settings){
+  return exports.viewedCheckoutStep(track, settings);
+};
+
+/**
+ * Map Completed Checkout Step.
+ *
+ * https://developers.google.com/analytics/devguides/collection/protocol/v1/devguide#enhancedecom
+ * https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters
+ *
+ * Required parameters:
+ *    - cid      (set in createcommonGAForm)
+ *    TODO: Is this option, or options?
```

Comment:
just one at a time, as I understand it. how we handle checkout step options needs to be looked at across client and server; i'll make a jira ticket for that @hankim813 can investigate further. Think we should move forward with this regardless